lib/XUnion.js

```diff
@@ -8,12 +8,13 @@ const P2P = require('./p2p/P2P');
 const P2PServer = require('./p2p/P2PServer');
 
 class XUnion {
-  constructor() {
+  constructor(args) {
+    this.args = args;
     this.logger = Logger.global;
   }
 
   async start() {
-    this.config = new Config();
+    this.config = new Config(this.args);
```

Comment:
Can you move `this.config = new Config(this.args);` to the constructor? That way we don't need to make `args` a class property. Otherwise looks good to squash and merge.